### DISCUSSION TITLE
Use Tortoise as a backend for a library without blocking usage?

### DISCUSSION THREAD

**Discussion by Cikmo** [tortoise/tortoise-orm#1276] [Q&A]
How can I, if possible, use tortoise in a library without blocking the user from using tortoise as well for other purposes?

**Comment by long2ice** [comment]
Detail?

**Reply by Cikmo** [reply]
well, since tortoise uses a global state, i don't really see how i could use it as a database backend for a library, and still be able to use it outside of the library.

Lets say I write an application. I import a library that uses Tortoise in the background. But now I want to use Tortoise for my application for other purposes.

How would I, as a library writer, allow this?

**Reply by Cikmo** [reply]
Is it possible to create an instance of a tortoise connection, for example? rather than global state?

**Comment by pkucmus** [comment]
I have a related question - how to use Tortoise in a library in general? Is that a considered option?
Right now I'm making a library that is to be used across a few applications. The library has a reporting component to it, the migration for the "Report" model along with the Aerich configuration lands in the pyproject.toml of the library - all good so far. But what if the application will also have it's own models? It does not seem that I could point to many `location` points in my app, or am I thinking about it wrong? 
I naturally think about all this like about Django apps but maybe it's not "the way"?
